File: week02/A-baloni/main.cpp
#include <iostream>
#include <vector>
using namespace std;
using ll = long long;

int main() {
    int n;
    cin >> n;
    int ans = 0;
    vector<int> arrows(1000003);
    while (n--) {
        int h;
        cin >> h;
        if (arrows[h] == 0) {
            ans++;
            arrows[h - 1]++;
        } else {
            arrows[h]--;
            arrows[h - 1]++;
        }
    }
    cout << ans << endl;
}

File: week02/A-baloni/main.rs
use std::io;

fn main() {
    let stdin = io::read_to_string(io::stdin()).unwrap();
    let mut stdin = stdin.split_whitespace();
    _ = stdin.next().unwrap();
    let h: Vec<usize> = stdin.map(|v| v.parse().unwrap()).collect();
    let mut arrows = vec![0; 1000001];
    let mut ans = 0;
    for h in h {
        if arrows[h] == 0 {
            ans += 1;
            arrows[h - 1] += 1;
        } else {
            arrows[h] -= 1;
            arrows[h - 1] += 1;
        }
    }
    println!("{}", ans);
}
